### PR TITLE
Add parameter in ReactorMeshParams to set block name in RGMB MG's based on region ID EEID

### PR DESCRIPTION
**Pull request by shikhar413**
## Impact

refs #27673

### CODE REVIEW

**Comment by shikhar413** [comment]
@eshemon FYI this is a feature I am adding to RGMB that should address the issue that the Griffin team raised that they need an automated way to set block names for each region ID in the problem, instead of collapsing all blocks into 2 names (one for tri and one for quad)

**Comment by miaoyinb** [comment]
I will try to review this PR as a practice. @lindsayad

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C [hunk 1]
```diff
@@ -671,6 +677,8 @@ CoreMeshGenerator::generate()
       auto elem_block_name = default_block_name;
       if (has_block_names)
         elem_block_name += "_" + _pin_block_name_map[pin_type_id][z_id][radial_idx];
+      if (getReactorParam<bool>(RGMB::region_id_as_block_name))
```
Comment (miaoyinb) [inline modules/reactor/src/meshgenerators/CoreMeshGenerator.C:680]: same

File: modules/reactor/src/meshgenerators/ReactorMeshParams.C [hunk 2]
```diff
@@ -43,6 +43,7 @@ ReactorMeshParams::validParams()
   params.addParam<std::vector<unsigned int>>(
       "axial_mesh_intervals",
       "Number of elements in the Z direction for each axial region");
+  params.addParam<bool>("region_id_as_block_name", false, "Set block name based on region id");
```
Comment (miaoyinb) [inline modules/reactor/src/meshgenerators/ReactorMeshParams.C:46]:
```suggestion
  params.addParam<bool>("region_id_as_block_name", false, "Set block names based on region IDs");
```

File: modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests [hunk 3]
```diff
@@ -268,6 +277,13 @@
     csvdiff = 'core_periphery_ptmg_in.csv'
     recover = false
   []
+  [periphery_rid_as_block_name_err]
+    requirement = 'The system shall throw an error when generating a core mesh with manually defined periphery block name and setting option to automatically define block names based on region IDs'
```
Comment (miaoyinb) [inline modules/reactor/test/tests/meshgenerators/core_mesh_generator/tests:281]:
```suggestion
    requirement = 'The system shall throw an error when generating a core mesh with a manually defined periphery block name and being set to automatically define block names based on region IDs'
```

File: modules/reactor/test/tests/meshgenerators/pin_mesh_generator/tests [hunk 4]
```diff
@@ -13,6 +13,31 @@
     exodiff = 'pin_only_in.e'
     recover = false
   []
+  [square_rid_as_block_name]
+    requirement = 'The system shall generate a 2D square pin mesh with fuel, gap, clad, background regions, with separate block names for each region ID in the mesh'
```
Comment (miaoyinb) [inline modules/reactor/test/tests/meshgenerators/pin_mesh_generator/tests:17]:
```suggestion
    requirement = 'The system shall generate a 2D square pin mesh with fuel, gap, clad, background regions, with separate block names assigned based on region IDs in the mesh'
```

File: modules/reactor/test/tests/meshgenerators/pin_mesh_generator/tests [hunk 5]
```diff
@@ -13,6 +13,31 @@
...
+                Mesh/pin1/ring_radii='0.3385 0.3705 0.4665'
+                Mesh/pin1/mesh_intervals='1 1 1 1'
+                Mesh/pin1/region_ids='1 2 3 4'
+                Mesh/pin1/num_sectors='4'
+                Outputs/file_base=pin_rid_as_block_name"
+    exodiff = 'pin_rid_as_block_name.e'
+    recover = false
+  []
+  [square_rid_as_block_name_err]
+    requirement = 'The system shall throw an error when generating a pin mesh with manually defined block names and setting option to automatically define block names based on region IDs'
```
Comment (miaoyinb) [inline modules/reactor/test/tests/meshgenerators/pin_mesh_generator/tests:30]:
```suggestion
    requirement = 'The system shall throw an error when generating a pin mesh with manually defined block names and being set to automatically define block names based on region IDs'
```

**Review by miaoyinb — CHANGES_REQUESTED** [review]
I finished my first round review. Thanks.

File: modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C [hunk 6]
```diff
@@ -551,6 +559,8 @@ AssemblyMeshGenerator::generate()
       auto elem_block_name = default_block_name;
       if (has_block_names)
         elem_block_name += "_" + _pin_block_name_map[pin_type_id][z_id][radial_idx];
+      else if (getReactorParam<bool>(RGMB::region_id_as_block_name))
```
[the remark this answers is not in the document]
Comment (miaoyinb) [reply modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C:562]: Yes, you are right

File: modules/reactor/src/meshgenerators/CoreMeshGenerator.C (hunk 1 above)
Comment (shikhar413) [reply modules/reactor/src/meshgenerators/CoreMeshGenerator.C]: I added an else here, if that is what you meant

File: modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C [hunk 7]
```diff
@@ -582,6 +592,8 @@ AssemblyMeshGenerator::generate()
 
       // Set element block name and block id
       auto elem_block_name = default_block_name;
+      if (getReactorParam<bool>(RGMB::region_id_as_block_name))
+        elem_block_name += "_REG" + std::to_string(elem_rid);
       if (is_background_region && _has_background_block_name)
```
[the remark this answers is not in the document]
Comment (shikhar413) [reply modules/reactor/src/meshgenerators/AssemblyMeshGenerator.C]: Good catch

**Comment by shikhar413** [comment]
@miaoyinb is this PR ready to merge?

**Comment by miaoyinb** [comment]
It looks okay to me. But @GiudGiud, @lindsayad or others may want to take another look?

**Review by lindsayad — APPROVED** [review]

**Pull request merged**